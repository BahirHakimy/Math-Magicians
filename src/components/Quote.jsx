import React from 'react';
import getQuote from '../api/quotes';

function Quote() {
  const [quote, setQuote] = React.useState({});
  const [loading, setLoading] = React.useState(false);
  const [error, setError] = React.useState(null);

  React.useEffect(() => {
    setLoading(true);
    getQuote()
      .then((quote) => {
        setQuote(quote);
        setLoading(false);
      })
      .catch(() => {
        setError('Failed to fetch the quote, please try again');
        setLoading(false);
      });
  }, []);

  return (
    <div
      style={{
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        height: '90vh',
        backdropFilter: 'blur(5px)',
        animation: 'appear 0.5s ease-in 1 forwards',
      }}
    >
      <div
        style={{
          display: 'flex',
          flexDirection: 'column',
          justifyContent: 'center',
          alignItems: 'center',
        }}
      >
        <h1
          style={{
            fontFamily: 'cursive',
            color: error ? 'orangered' : '#fff',
            filter: 'drop-shadow(0.2rem 0.2rem 0.3rem #77c)',
            textShadow: ' 0.1rem 0.1rem 0.1rem black',
            padding: '1rem 2rem',
            textAlign: 'center',
          }}
        >
          {error || (loading ? 'Loading...' : quote[0]?.quote)}
        </h1>
        <cite
          style={{
            padding: '1rem',
            marginTop: '1rem',
            color: ' #00a1ff',
            filter: 'drop-shadow(0.2rem 0.2rem 0.3rem #000)',
            textShadow: '0.1rem 0.1rem 0.5rem #333',
          }}
        >
          {!loading && !error ? quote[0]?.author : ''}
        </cite>
      </div>
    </div>
  );
}

export default Quote;
