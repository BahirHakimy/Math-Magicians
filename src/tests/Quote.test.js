import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Quote from 'components/Quote';

describe('Unit tests for Quote component', () => {
  test('A Loading text should be rendered in the firstplace', () => {
    render(<Quote />);
    expect(screen.getByText('Loading...')).toBeInTheDocument();
  });

  test('Quote component should be rendered correctly with all its elements', () => {
    const quote = render(<Quote />);
    expect(quote).toMatchSnapshot();
  });
});
