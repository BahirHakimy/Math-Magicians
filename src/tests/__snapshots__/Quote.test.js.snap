// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for Quote component Quote component should be rendered correctly with all its elements 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: flex; justify-content: center; align-items: center; height: 90vh; animation: appear 0.5s ease-in 1 forwards;"
      >
        <div
          style="display: flex; flex-direction: column; justify-content: center; align-items: center;"
        >
          <h1
            style="font-family: cursive; color: rgb(255, 255, 255); filter: drop-shadow(0.2rem 0.2rem 0.3rem #77c); text-shadow: 0.1rem 0.1rem 0.1rem black; padding: 1rem 2rem; text-align: center;"
          >
            Loading...
          </h1>
          <cite
            style="padding: 1rem; margin-top: 1rem; color: rgb(0, 161, 255); filter: drop-shadow(0.2rem 0.2rem 0.3rem #000); text-shadow: 0.1rem 0.1rem 0.5rem #333;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex; justify-content: center; align-items: center; height: 90vh; animation: appear 0.5s ease-in 1 forwards;"
    >
      <div
        style="display: flex; flex-direction: column; justify-content: center; align-items: center;"
      >
        <h1
          style="font-family: cursive; color: rgb(255, 255, 255); filter: drop-shadow(0.2rem 0.2rem 0.3rem #77c); text-shadow: 0.1rem 0.1rem 0.1rem black; padding: 1rem 2rem; text-align: center;"
        >
          Loading...
        </h1>
        <cite
          style="padding: 1rem; margin-top: 1rem; color: rgb(0, 161, 255); filter: drop-shadow(0.2rem 0.2rem 0.3rem #000); text-shadow: 0.1rem 0.1rem 0.5rem #333;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
